src/graph/network.cc

```diff
@@ -24,7 +24,32 @@ using namespace dgl::runtime;
 namespace dgl {
 namespace network {
 
-void MsgMeta::AddArray(const NDArray& array) {
+NDArray CreateNDArrayFromRaw(std::vector<int64_t> shape,
+                             DLDataType dtype,
+                             DLContext ctx,
+                             void* raw) {
+  DLTensor tensor;
+  tensor.ctx = ctx;
+  tensor.ndim = static_cast<int>(shape.size());
+  tensor.dtype = dtype;
+  tensor.shape = new int64_t[tensor.ndim];
+  for (int i = 0; i < tensor.ndim; ++i) {
+    tensor.shape[i] = shape[i];
+  }
+  tensor.strides = new int64_t[tensor.ndim];
+  for (int i = 0; i < tensor.ndim; ++i) {
+    tensor.strides[i] = 1;
+  }
+  for (int i = tensor.ndim - 2; i >= 0; --i) {
+    tensor.strides[i] = tensor.shape[i+1] * tensor.strides[i+1];
+  }
+  tensor.data = raw;
+  DLManagedTensor *managed_tensor = new DLManagedTensor();
+  managed_tensor->dl_tensor = tensor;
+  return NDArray::FromDLPack(managed_tensor);
+}
```

Comment:
No NDArray constructor creates an array in this way?

Reply:
I suppose the created array doesn't own the memory. @jermainewang is this a right way?